File: cache/src/main/java/com/mm/http/cache/CacheConverter.java
package com.mm.http.cache;


import com.mm.http.RetrofitCache;
import javax.annotation.Nullable;
import retrofit2.Response;

/**
 * Created by : majian
 * Date : 2021/8/24
 * CacheConverter
 */

public interface CacheConverter<T> {
    @Nullable
    Response<T> convert(Response<T> response);

    /**
     * Creates {@link CacheConverter} instances based on a type and target usage.
     */
    abstract class Factory {

        @Nullable
        public CacheConverter<?> converterCache(RetrofitCache retrofit) {
            return null;
        }
    }
}

File: cache/src/main/java/com/mm/http/cache/CacheConverter.kt
package com.mm.http.cache

import com.mm.http.RetrofitCache
import retrofit2.Response

/**
 * Created by : majian
 * Date : 2021/9/30
 */

interface CacheConverter<T> {
    fun convert(response: Response<T>): Response<T>?

    /**
     * Creates [CacheConverter] instances based on a type and target usage.
     */
    abstract class Factory {

        open fun <T> converterCache(retrofit: RetrofitCache): CacheConverter<T>? {
            return null
        }
    }
}
